lib/bsdlib/bsdlib.c

```diff
@@ -42,8 +121,12 @@ static int _bsdlib_init(struct device *unused)
 	return init_ret;
 }
 
+
 int bsdlib_init(void)
 {
+#ifdef CONFIG_BSD_LIBRARY_ALLOW_SECURE
+	setup_for_secure();
```

Comment:
Curious, why is this needed?

Reply:
If its more then 1 line just explain it IRL xD

Reply:
It's needed to setup the clock(BSDLib tries to enable the LF clock if it's not enabled) the modem is similar to nRF53 where it requires the RAM section used to be setup as non-secure(if not it won't be addressable on the BUS since the Modem is always a non-secure domain) Same with power it needs to be set as NS since BSDLib uses NS addressing same with the IPC. I think maybe we could allow POWER to be split. Sorry for the magic values seems they got reverted at some point.

So basically:
IPC NS
POWER NS or SPLIT
RAM NS

Are the requirements for operating with the modem from the secure domain.

We could also make
CLOCK NS and maybe it would work then we don't have to enable the LF clock and BSDLib could maybe handle it correctly if we allow Non-secure mapping. I'll have to test that.